controllers/front/landing.php

```diff
@@ -83,20 +87,24 @@ private function renderProductList()
             $this->context->getTranslator()
         );
 
-        $products = [];
-        foreach ($this->products as $productDetail) {
-            $products[] = $presenter->present(
-                $presentationSettings,
-                $assembler->assembleProduct($productDetail),
-                $this->context->language
-            );
+        $blocks_count = 0;
+        foreach ($this->products as $product) {
+            $products = [];
+            foreach ($product as $productDetail) {
+                $products[] = $presenter->present(
+                    $presentationSettings,
+                    $assembler->assembleProduct($productDetail),
+                    $this->context->language
+                );
+            }
+            $this->landing_data['blocks'][$blocks_count]['products'] = $products;
+            ++$blocks_count;
```

Comment:
Así quitas la magia esa que intentas hacer el con el contador (ver también linea 47)
```suggestion
        foreach ($this->landing_data['blocks'] as &$block) {
            $products = [];
            foreach ($block['products'] as $productDetail) {
                $products[] = $presenter->present(
                    $presentationSettings,
                    $assembler->assembleProduct($productDetail),
                    $this->context->language
                );
            }
           $block['products'] = $products;
```

Reply:
Me parece una idea increíble, muchísimas gracias :smile: